### PR TITLE
MULE-9023: Scatter-gather generates wrong data type when Content-Type…

### PR DESCRIPTION
**Pull request by pablokraan**
… header is present

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpRequestToMuleEvent.java [hunk 1]
```diff
@@ -122,6 +122,15 @@ else if (entity instanceof InputStreamHttpEntity)
         }
 
         final DefaultMuleMessage defaultMuleMessage = new DefaultMuleMessage(payload, inboundProperties, outboundProperties, inboundAttachments, muleContext);
+
+        for (String property : inboundProperties.keySet())
+        {
+            if (property.equalsIgnoreCase(MuleProperties.CONTENT_TYPE_PROPERTY))
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpRequestToMuleEvent.java]: Use org.mule.module.http.api.HttpHeaders.Names#CONTENT_TYPE instead.

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpRequestToMuleEvent.java [hunk 2]
```diff
@@ -122,6 +122,15 @@ else if (entity instanceof InputStreamHttpEntity)
         }
 
         final DefaultMuleMessage defaultMuleMessage = new DefaultMuleMessage(payload, inboundProperties, outboundProperties, inboundAttachments, muleContext);
+
+        for (String property : inboundProperties.keySet())
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpRequestToMuleEvent.java]: Why not use "request.getHeaderValue(HttpHeaders.Names.CONTENT_TYPE);" as used already on line 86.  Of move line 86 up so it's in scope?

File: modules/http/src/main/java/org/mule/module/http/internal/request/HttpResponseToMuleEvent.java [hunk 3]
```diff
@@ -108,9 +108,18 @@ else if (responseContentType.startsWith(APPLICATION_X_WWW_FORM_URLENCODED.toLowe
         }
 
 
-        MuleMessage message = new DefaultMuleMessage(muleEvent.getMessage().getPayload(), inboundProperties,
+        DefaultMuleMessage message = new DefaultMuleMessage(muleEvent.getMessage().getPayload(), inboundProperties,
                                                      null, inboundAttachments, muleContext, muleEvent.getMessage().getDataType());
 
+
+        for (String property : inboundProperties.keySet())
```
Comment (dfeist) [inline modules/http/src/main/java/org/mule/module/http/internal/request/HttpResponseToMuleEvent.java]: This is the second time i've seen this right after DefaultMuleMessage constructor, is there no way it can be part of the constructor logic?

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpRequestToMuleEvent.java (hunk 2 above)
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpRequestToMuleEvent.java]: Yes, refactoring the code to make use of that line is out of scope (there are 4 levels of "if" in that code, I prefer to keep them untouched)

File: modules/http/src/main/java/org/mule/module/http/internal/request/HttpResponseToMuleEvent.java (hunk 3 above)
Comment (pablokraan) [inline modules/http/src/main/java/org/mule/module/http/internal/request/HttpResponseToMuleEvent.java]: Right, this code is duplicated. I can move it to some static method, but moving it to the constructor will introduce the original problem again
